File: from-2201-to-2300/2285-maximum-total-importance-of-roads/solution.py
class Solution:
    def maximumImportance(self, n: int, roads: List[List[int]]) -> int:
        edgeCount = [0] * n
        for c1, c2 in roads:
            edgeCount[c1] += 1
            edgeCount[c2] += 1

        result = 0
        label = 1
        for count in sorted(edgeCount):
            result += label * count
            label += 1

        return result

File: from-2201-to-2300/2285-maximum-total-importance-of-roads/solution.go
func maximumImportance(n int, roads [][]int) int64 {
    edgeCount := make([]int, n)
    for _, r := range roads {
        edgeCount[r[0]] += 1
        edgeCount[r[1]] += 1
    }
    sort.Ints(edgeCount)
    var result int64 = 0
    label := 1
    for _, count := range edgeCount {
        result += int64(label * count)
        label += 1
    }
    return result
}